### DISCUSSION TITLE
Control ControlNet with a prompt keyword

### DISCUSSION THREAD

**Discussion by anonghuser** [Mikubill/sd-webui-controlnet#1598] [Ideas]
Would it be possible to assign a keyword, or multiple ones for each multi-controlnet config, and have them enabled or disabled depending on the prompt? It could be hardcoded ones such as `controlnet` (for all), `controlnet2` (for a specific one), etc. or user-assignable in the configs, as you think makes more sense.

Several use cases I can think of:
- assign a specific reference-only control to different regions when combined with regional prompter 
- finer control of batching with dynamic prompt generation
- the "prompt is more important" vs "controlnet is more important" option becomes redundant, users can just specify weight for controlnet's keyword and the rest of the prompt [maybe?]

Bonus points if some parameters can be changed from the prompt with a special syntax, i.e. `(controlnet3:0.6:preprocessor=none:filename=/another/image.png)` but if that's too wild then at least consider the basic use cases with just 0/1 boolean toggle or fraction weight where makes sense.

**Comment by ljleb** [comment]
There was a discussion earlier about making controlnet controllable from the text prompt, I need to find it. I am not sure anymore exactly how useful this would be or how easy it would be to integrate it with other extensions.
For prompt keywords, capital letters are tokenized the same way as lower case letters. This means we can reserve any capital word for our own prompt parsing/preprocessing needs.

I don't have a good example, but i.e. you could write `CNET(weight: 0.3, model: depth, ...)` to enable unit 0 from the prompt.
